gtars/src/uniwig/mod.rs

```diff
@@ -169,13 +186,22 @@ fn clamped_start_position(start: i32, smoothsize: i32) -> i32 {
 /// Main function
 pub fn uniwig_main(
     smoothsize: i32,
-    combinedbedpath: &str,
+    filepath: &str,
     chromsizerefpath: &str,
     bwfileheader: &str,
     output_type: &str,
+    filetype: &str,
 ) -> Result<(), Box<dyn Error>> {
-    let stepsize = 1;
+    // Determine File Type
+    let ft = match filetype.to_lowercase().as_str() {
+        "bed" => Ok(FileType::BED),
+        "bam" => Ok(FileType::BAM),
+        _ => Err(format!("Invalid file type: {}", filetype)),
+    };
 
+    //println!("Supplied file type: {:?}", ft.unwrap());
+
+    let stepsize = 1;
     // Set up output file names
 
     let mut file_names: [String; 3] = [
```

Comment:
This variable is defined as `mut` but never actually is mutated... suggest removing to reduce overhead

Reply:
Actually the variable is never used at all